borgmatic/hooks/mysql.py

```diff
@@ -105,13 +109,20 @@ def execute_dump_command(
     if dry_run:
         return None
 
-    dump.create_named_pipe_for_dump(dump_filename)
-
-    return execute_command(
-        dump_command,
-        extra_environment=extra_environment,
-        run_to_completion=False,
-    )
+    if is_directory_format:
+        dump.create_parent_directory_for_dump(dump_filename)
+        execute_command(
+            dump_command,
+            extra_environment=extra_environment,
+        )
+        return None
+    else:
```

Comment:
You could lose the `else:` given that the other code path returns early. (Do not feel strongly.)